client/src/main/java/io/getstream/chat/android/client/helpers/AttachmentHelper.kt

```diff
@@ -0,0 +1,31 @@
+package io.getstream.chat.android.client.helpers
+
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.client.utils.SystemTimeProvider
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+import org.apache.commons.validator.routines.UrlValidator
+
+class AttachmentHelper(private val systemTimeProvider: SystemTimeProvider = SystemTimeProvider()) {
+
+    private val urlValidator = UrlValidator()
+
+    fun hasValidUrl(attachment: Attachment): Boolean {
+        val url = attachment.url
+        return when {
+            url == null -> false
+            urlValidator.isValid(url).not() -> false
+            url.contains(QUERY_KEY_NAME_EXPIRES) -> {
+                parseTimeStampOrNull(url)?.let { timestamp -> timestamp > systemTimeProvider.provideTime() } ?: false
+            }
+            else -> true
+        }
+    }
+
+    private fun parseTimeStampOrNull(url: String): Long? {
+        return url.toHttpUrlOrNull()?.queryParameter(QUERY_KEY_NAME_EXPIRES)?.toLongOrNull()
+    }
```

Comment:
Since we started using `toHttpUrlOrNull` from OkHttp, which as I understand validates the URL while parsing, we could simplify all the way to something like this:

```suggestion
    fun hasValidUrl(attachment: Attachment): Boolean {
        val timeStamp = attachment.url
            ?.toHttpUrlOrNull() 
            ?.queryParameter(QUERY_KEY_NAME_EXPIRES)
            ?.toLongOrNull()
            ?: return false
        return timeStamp > systemTimeProvider.provideTime()
    }
```

This would also let us drop the URL validator dependency.

Reply:
Your code seems very elegant, but unfortunately it fails test when url doesn't contain Query param "expires" (I'm not sure, maybe such url without expires should not be valid). But you're right we can omit validator dependency

Reply:
Oops, that's correct, I missed that case when writing this up. Nicely done with the tests that they caught it 👍